### PR TITLE
Give the God's Chosen NPCs some anti-mutant dialogue

### PR DESCRIPTION
**Pull request by MNG-cataclysm**
#### Summary
Content "Give the God's Chosen NPCs some anti-mutant dialogue"

#### Purpose of change
The God's Chosen are described as being very anti-mutant in their beliefs, yet this was not at all modeled within the game. This PR aims to change that and flesh them out some more.

#### Describe the solution
Gave several of the NPC's anti-mutant dialogue, many outright refusing to speak with the player if they are a post-threshold mutant. Some members of the sect, particularly those who are not full members or haven't completely bought into it, will still speak with the player on normal terms, and have unique dialogue which addresses this.

#### Describe alternatives you've considered
Currently, the EOC which determines the no talk value only activates if you are a post-threshold mutant. I have considered using this on top of a certain accumulated mutation "visibility", the same effect triggering if the player is visibly mutated in a significant fashion.
If you lose your mutation threshold by some means, the no talk value goes away, and the sect can be interacted with normally. I assumed that most of the members would not recognize you between your mutated and un-mutated self, though I'm not sure if this needs to be changed.
I have also considered having the NECC become automatically hostile to you through the mutated means, but this makes several non-xenophobic NPCs hostile for no good reason. If you defend yourself, this also permanently locks you out of any interaction with the group besides attacking.

#### Testing
I mutated post-threshold and walked around the compound. All the dialogue worked as a planned. I removed my mutation, and could talk with the members normally.

#### Additional context
None at this time, section may change later.

### CODE REVIEW

File: data/json/npcs/godco/foodguard.json [hunk 1]
```diff
@@ -63,7 +110,7 @@
   {
     "type": "talk_topic",
     "id": "TALK_GODCO_guard_food_borrow",
-    "dynamic_line": [ "Nope, you can't do that.  I'd like to help you, but we've already got enough mouths to feed." ],
+    "dynamic_line": [ "Nope, you can't do that.  I'd like to help you, bit we've already got enough mouths to feed." ],
```
Comment (MNG-cataclysm) [inline data/json/npcs/godco/foodguard.json:113]:
```suggestion
    "dynamic_line": [ "Nope, you can't do that.  I'd like to help you, but we've already got enough mouths to feed." ],
```

File: data/json/npcs/godco/godco_missions.json [hunk 2]
```diff
@@ -510,7 +510,7 @@
     "has_generic_rewards": false,
     "dialogue": {
       "describe": "I have a job for you…",
-      "offer": "As a matter of fact, yes you can.  You see, we're running low on toilet paper.  It's not something we can't live without, but it's luxury we can afford.  I can pay you handsomely if you find us ten rolls of toilet paper.",
+      "offer": "As a matter of fact, yes you can.  You see, we're running low on toilet paper.  It's not something we can't live without, but it's luxury we can afford.  I can pay you handsomely if you find us ten rolls toilet paper.",
```
Comment (MNG-cataclysm) [inline data/json/npcs/godco/godco_missions.json:513]:
```suggestion
      "offer": "As a matter of fact, yes you can.  You see, we're running low on toilet paper.  It's not something we can't live without, but it's luxury we can afford.  I can pay you handsomely if you find us ten rolls of toilet paper.",
```

File: data/json/npcs/godco/NPC_Corrie_Kaja_Dosia.json [hunk 3]
```diff
@@ -225,5 +248,14 @@
     "id": "TALK_GODCO_Corrie_Recruit",
     "dynamic_line": "Sure, I'll come.  I hate staying here, and you've gone out of your way to help me.  I appreciate that, <name_g>.",
     "responses": [ { "text": "Let's hit the road.", "topic": "TALK_DONE", "effect": "follow" } ]
+  },
+  {
+    "type": "talk_topic",
+    "id": "TALK_GODCO_Corrie_You_Mutant",
+    "dynamic_line": "Nearly everyone here went to a church that was a bit… out there, shall we say.  They hate what they call \"mutants\", how some creatures and people have physically changed inexplicablly after everything went to shit.  I'm glad they didn't kill you on the spot, not many people are left out there.",
```
Comment (MNG-cataclysm) [inline data/json/npcs/godco/NPC_Corrie_Kaja_Dosia.json:255]:
```suggestion
    "dynamic_line": "Nearly everyone here went to a church that was a bit… out there, shall we say.  They hate what they call \"mutants\", how some creatures and people have physically changed inexplicably after everything went to shit.  I'm glad they didn't kill you on the spot, not many people are left out there.",
```

**Comment by anoobindisguise** [comment]
They have no way to tell if you're an alpha mutant for example. I don't know if thresholding is the right way to go about it. Going by total visibility of stuff/ugliness is probably better, maybe with three tiers of dialogue? That's a lot of work, so I understand if you don't wanna do that.
Ideally there'd be "1-3 mutations like Makayla or Yusuke" "Several mutations" and "Abomination" maybe?

**Comment by MNG-cataclysm** [comment]
> They have no way to tell if you're an alpha mutant for example. I don't know if thresholding is the right way to go about it. Going by total visibility of stuff/ugliness is probably better, maybe with three tiers of dialogue? That's a lot of work, so I understand if you don't wanna do that.
> 
> Ideally there'd be "1-3 mutations like Makayla or Yusuke" "Several mutations" and "Abomination" maybe?

That sounds like a good idea. However, since they consider any visible mutation a sign of judgement by God against the person, I guess we could use the same dialogue for the first two tiers, and more extreme cases for the third?

**Comment by fairyarmadillo** [comment]
Going by ugliness is the best method, and we also should make the guards at the refugee center turn you away if you're too badly mutated. Maybe they'd still let you in the entryway to talk to Smokes.

**Comment by andrei8l** [comment]
> They have no way to tell if you're an alpha mutant for example.

The player can recognize the alpha children in the TCL as mutant so there must be *something* visible, even if it's just a vague uncanny aspect.

**Comment by BlizzDA** [comment]
> The player can recognize the alpha children in the TCL as mutant so there must be _something_ visible, even if it's just a vague uncanny aspect.
Always up to discussion in the end, but the child's strange behavior, as noted in the description, and the fact it's located in same place allied to mutants should be enough of context clues imo. It's also wearing the same jumpsuits as the other subjects.

**Comment by BlizzDA** [comment]
> > The player can recognize the alpha children in the TCL as mutant so there must be _something_ visible, even if it's just a vague uncanny aspect.

Always up to discussion in the end, but the child's strange behavior, as noted in the description, and the fact it's located in same place allied to mutants should be enough of context clues imo. It's also wearing the same jumpsuits as the other subjects.

**Comment by Procyonae** [comment]
Ugliness seems like a weird way to do this imo, that means like an Elf-A mutant would still be fine, wouldn't like the sum of absolute mutation point costs be better? That way it doesn't matter if you're thresholded or only have good or bad mutations

**Comment by fairyarmadillo** [comment]
> > They have no way to tell if you're an alpha mutant for example.
> 
> The player can recognize the alpha children in the TCL as mutant so there must be _something_ visible, even if it's just a vague uncanny aspect.

That child, like the other mutants in the room, is far more mutated than the player can ever get. Here's their description:

A thin child with a shaved head, wearing a too-large white lab coat over a baggy orange jumpsuit. They turn their head towards you with uncanny speed, and you're takеn aback by the inhuman curiosity burning in their eyes. Their angelic smile does very little to reassure you of their intentions.

and the transcendant mutant, which again is far beyond even the threshold:

An experimental subject wearing a surprisingly clean orange jumpsuit. They look unchanged at a distance, but as you approach their head spins towards you with unnatural speed. Their face is disconcertingly beautiful, but their eyes shine with an intelligence far beyond human - and a deep disgust directed at you.

None of the Alpha mutations give you ugliness. None of the alpha mutations except for the upper two beauty ones take you beyond normal human capabilities. I suppose it could be argued that a person in those upper echelons might look a bit uncanny, but you'd have to know what you were looking for to ID that as a mutant, and given that 1) Alpha was a super secret mutagen developed at a single lab one week before things went to shit and 2) the person in question has such massive social advantages due to being more human than human, they'd probably easily be able to convince you otherwise, unlike the various Brundleflies and Wolfmans wandering the wasteland.

The original Alpha proposals outlined it as a super-soldier serum whose users would be able to convince people that they weren't even really mutants. The plan was to roll it out slowly to open the public up to the idea of mutagens before doing anything more exotic with it. So unless these people have inside knowledge about XEDRA's most secret plans, they would probably just think you were a suspiciously good looking survivor.

**Comment by fairyarmadillo** [comment]
> Ugliness seems like a weird way to do this imo, that means like an Elf-A mutant would still be fine, wouldn't the sum of absolute mutation point costs be better? That way it doesn't matter if you're thresholded or only have good or bad mutations

But again a thresholded alpha would be totally fine, and some mutants are completely hideous well before the threshold, like slime, snail, and insect.

There are a few other options: 

* split out beauty and visible mutatedness into separate stats
* just go off beauty but also check for a few specific mutations
* automatically count thresholded mutants as sus (which could include alpha I suppose), and use ugliness to determine whether they're completely shunned

**Comment by BlizzDA** [comment]
Re: Elf-A getting a pass via not being hideous to look at. Lore-wise, might there be a little room for faction variety there? Exodii are slated to shun all sufficiently mutated individuals. Perhaps it could be the same here for most but the beautiful Elf-A is taken as some sort of 'new paradise on Earth' creature? So it and human-passable mutations can still interact with the community, without requiring you to switch to an allied NPC.

**Comment by SariusSkelrets** [comment]
Are they intended to talk about the rapture?

AFAIK they call it "the second flood" due to considering that it's impossible that God would leave them behind during the rapture

**Comment by fairyarmadillo** [comment]
> Re: Elf-A getting a pass via not being hideous to look at. Lore-wise, might there be a little room for faction variety there? Exodii are slated to shun all sufficiently mutated individuals. Perhaps it could be the same here for most, but the beautiful Elf-A is taken as some sort of 'new paradise on Earth' creature? So it and human-passable mutations can still interact with the community, without requiring you to switch to an allied NPC.

I think it would depend on the faction. An elf could probably talk their way into the refugee center, but being pretty won't sway most people's religious convictions, they'd just call you a demonic temptress or something.

**Comment by MNG-cataclysm** [comment]
> Are they intended to talk about the rapture?
> 
> 
> 
> AFAIK they call it "the second flood" due to considering that it's impossible that God would leave them behind during the rapture

They aren't intended to talk about the Rapture, I'll fix that.

**Comment by MNG-cataclysm** [comment]
Regarding the use of mutation visibility in place of thresholding; I tried using the "visibility" entry listed in the mutations doc as a conditional, but it returned bad, and I'm still searching for an example of where it's used in dialogue as a condition. If it works, I'll switch to that.

**Comment by TrueChevaleresse** [comment]
Worth noting that attractiveness mutations are highly visible IIRC but not really outside human bounds as described, other than maybe Glorious and Grotesque. It wouldn't really make sense for people to decry you as an abomination against God for being hot (or ugly.)

**Comment by MNG-cataclysm** [comment]
> Worth noting that attractiveness mutations are highly visible IIRC but not really outside human bounds as described, other than maybe Glorious and Grotesque. It wouldn't really make sense for people to decry you as an abomination against God for being hot (or ugly.)

This current system doesn't use attractiveness as a conditional factor, just if you've passed a threshold. I'm going to experiment around with the new code support here soon.

**Comment by MNG-cataclysm** [comment]
Closing this in favor of #67525.

**Pull request closed without merging**